client/payment-methods/index.js

```diff
@@ -285,12 +288,20 @@ const PaymentMethods = () => {
 								allows_manual_capture: isAllowingManualCapture,
 								setup_required: isSetupRequired,
 								setup_tooltip: setupTooltip,
+								allows_pay_later: isAllowingPayLater,
 							} ) => (
 								<PaymentMethod
 									id={ id }
 									key={ id }
 									label={ label }
-									description={ description }
+									description={
+										isAllowingPayLater
+											? sprintf(
+													description,
+													stripeAccountDomesticCurrency.toUpperCase()
+											  )
+											: description
+									}
```

Comment:
The logic of constructing the description for BNPLs is repeated in several places. Maybe it's worth refactoring `paymentMethodsMap` and make the `description` value a function so we can call `paymentMethodsMap['description'](domesticCurrency)`? 

But please evaluate this suggestion. Could be I'm overengineering things now.

Reply:
Yes thanks, I made a refactor to centralize the logic to generate the description but following a different approach. Please let me know what you think: [10fada1](https://github.com/Automattic/woocommerce-payments/pull/6920/commits/10fada1029c3b936ee9f368e0830c9b25a3bc5fb)